lib/rucio/client/downloadclient.py

```diff
@@ -571,16 +573,17 @@ def _download_item(self, item, trace, traces_copy_out, log_prefix=''):
                 end_time = time.time()
 
                 if success and not item.get('merged_options', {}).get('ignore_checksum', False):
-                    rucio_checksum = item.get('adler32')
-                    local_checksum = None
-                    if rucio_checksum is None:
-                        rucio_checksum = item.get('md5')
-                        if rucio_checksum is None:
-                            logger.warning('%sNo remote checksum available. Skipping validation.' % log_prefix)
-                        else:
-                            local_checksum = md5(temp_file_path)
-                    else:
-                        local_checksum = adler32(temp_file_path)
+                    rucio_checksum = 'rucio_checksum'
+                    local_checksum = 'local_checksum'
+
+                    for checksum_name in GLOBALLY_SUPPORTED_CHECKSUMS:
+                        rucio_checksum_temp = item.get(checksum_name)
+                        if rucio_checksum_temp is not None:
+                            rucio_checksum = rucio_checksum_temp
+                            if checksum_name in CHECKSUM_ALGO_DICT:
+                                local_checksum = CHECKSUM_ALGO_DICT[checksum_name](temp_file_path)
+                                if checksum_name == PREFERRED_CHECKSUM:
+                                    break
```

Comment:
I suggest we create a method for the verification like this:
```
def _verify_checksum(item, file_path):
    rucio_checksum = item.get(PREFERRED_CHECKSUM)
    checksum_algo = CHECKSUM_ALGO_DICT.get(PREFERRED_CHECKSUM)
    if rucio_checksum and checksum_algo:
        return rucio_checksum == checksum_algo(file_path)

    for checksum_name in GLOBALLY_SUPPORTED_CHECKSUM:
        rucio_checksum = item.get(checksum_name)
        checksum_algo = CHECKSUM_ALGO_DICT.get(checksum_name)
        if rucio_checksum and checksum_algo:
            return rucio_checksum == checksum_algo(file_path)

    return False
```
and then just do `success = self._verify_checksum(item, temp_file_path)`.

Reply:
Sounds good to me, I'll do that by the end of the weekend!